packages/daemons/blocks_collection.go

```diff
@@ -87,7 +87,16 @@ func InitialLoad(logger *log.Entry) error {
 	return nil
 }
 
+var bcOnRun bool
+
 func blocksCollection(ctx context.Context, d *daemon) (err error) {
+	if bcOnRun {
+		return nil
+	}
+	defer func() {
+		bcOnRun = false
+	}()
+	bcOnRun = true
```

Comment:
It's not thread safe. Need to use sync/atomic.

Reply:
Fixed